### PR TITLE
Fix stratum timer memory leak

### PR DESCRIPTION
**Pull request by jean-m-cyr**
Kept allocating new timers without ever deleting them.

### CODE REVIEW

**Comment by chfast** [comment]
Please use unique_ptr or simply compose by value.

**Comment by jean-m-cyr** [comment]
Unique pointer will work, but add the overhead of constantly deleting and reallocating the timer.

I'll change it to a unique pointer, but refrain from reallocating unnecessarily.

File: libstratum/EthStratumClient.cpp [hunk 1]
```diff
@@ -43,14 +43,15 @@ EthStratumClient::EthStratumClient(Farm* f, MinerType m, string const & host, st
 	m_pending = 0;
 	m_maxRetries = retries;
 	m_worktimeout = worktimeout;
+	p_worktimer = unique_ptr<boost::asio::deadline_timer>(new boost::asio::deadline_timer(m_io_service, boost::posix_time::seconds(m_worktimeout)));
```
Comment (jean-m-cyr) [inline libstratum/EthStratumClient.cpp]: Is there a way to new a deadline_timer without arming it?

File: libstratum/EthStratumClient.cpp [hunk 2]
```diff
@@ -28,7 +28,8 @@ static void diffToTarget(uint32_t *target, double diff)
 
 
 EthStratumClient::EthStratumClient(Farm* f, MinerType m, string const & host, string const & port, string const & user, string const & pass, int const & retries, int const & worktimeout, int const & protocol, string const & email)
-	: m_socket(m_io_service)
+        :       m_socket(m_io_service),
+	        m_worktimer(m_io_service, boost::posix_time::seconds(99999)) // effectively set it to infinity
```
Comment (chfast) [inline libstratum/EthStratumClient.cpp]: 99999 not needed. The constructor without any time will not start the timer. See: http://www.boost.org/doc/libs/1_66_0/doc/html/boost_asio/reference/basic_deadline_timer/basic_deadline_timer/overload1.html.

Comment (jean-m-cyr) [reply libstratum/EthStratumClient.cpp]: Ah! Ok, I'll change that

**Pull request merged**